src/lib/mina_base/snapp_generators.ml

```diff
@@ -237,22 +237,34 @@ let gen_predicate_from ?(succeed = true) ~account_id ~ledger () =
             else return (Party.Predicate.Nonce (Account.Nonce.succ nonce)) )
 
 let gen_fee (account : Account.t) =
-  Currency.Fee.gen_incl Mina_compile_config.minimum_user_command_fee
-    Currency.(Amount.to_fee (Balance.to_amount account.balance))
+  let lo_fee = Mina_compile_config.minimum_user_command_fee in
+  let hi_fee = Currency.(Amount.to_fee (Balance.to_amount account.balance)) in
+  Currency.Fee.gen_incl lo_fee hi_fee
 
 let fee_to_amt fee = Currency.Amount.(Signed.of_unsigned (of_fee fee))
 
 let gen_delta ?balances_tbl (account : Account.t) =
-  let pk = account.public_key in
   let open Quickcheck.Let_syntax in
-  match%bind Quickcheck.Generator.of_list [ Sgn.Pos; Neg ] with
+  let pk = account.public_key in
+  (* TODO: the authorization is Signature for new accounts only
+     this will change when we split party's into two
```

Comment:
we could also add snapp accounts to the ledger. The accounts could have a dummy verification key and generated permissions. Parties can then be generated based on these permissions. We'll want snapp transactions on existing accounts anyway.

Reply:
I think we'll want to be careful about setting permissions at account creation time. There's a plausible attack where a block producer may
* see that there's going to be a new account created by a transaction,
* inserts a transaction before it that creates the account with permissions `{send: Either; ...}` and some verification key
* executes the transaction that originally would have created the account
* generates a transaction that steals the funds using a proof.

For this reason, 'setting up' an account should always require a signature for its public key. If we also want to provide an 'initialisation proof' for the snapp, that will need to be a separate party, because we don't support simultaneous auth by signature and proof.

Reply:
I was suggesting we do it only for tests so that we can test the updates that doesn't require changing the permissions in the same transaction